qiskit/transpiler/passes/mapping/apply_layout.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+A pass for transforming a circuit with virtual qubits into a circuit with physical qubits.
+"""
+from qiskit.circuit import QuantumRegister
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.transpiler.coupling import CouplingMap
+from qiskit.transpiler.layout import Layout
+
+
+class ApplyLayout(TransformationPass):
+    """
+    Transform a DAGCircuit with virtual qubits into a DAGCircuit with physical qubits.
+    """
+
+    def __init__(self, coupling: CouplingMap, initial_layout: Layout = None):
+        """
+        Transform a DAGCircuit with virtual qubits into a DAGCircuit with physical qubits
+        defined in the coupling by applying a given `initial_layout`.
+        Args:
+            coupling: initial layout to be applied
+            initial_layout: initial layout to be applied
+        """
+        super().__init__()
+        self._coupling = coupling
+        self._initial_layout = initial_layout
```

Comment:
It might be better to move the setting of `initial_layout` to it's own pass. That way, there's a consistent stream of "select layout" (one of TrivialLayout, DenseLayout, UserLayout, ...) into "apply_layout". Also, there is no way for an `initial_layout` provided here to make it into the `property_set`. Thoughts?

Reply:
That's a good point. I understand the necessity of improvement, and we have two options for it:

Option A: Remove `initial_layout` from argument of `__init__`. Users always use some of the Layout passes to set their layout to `property_set`. ApplyLayout apply the layout set in `property_set`. When users just want to use their own valid layout, they need to run kind of `SetLayout` pass before ApplyLayout pass. 

Option B: Add the following lines in `run()`.
```
        if self._initial_layout:
            self.property_set["layout"] = self._initial_layout

```
This means `ApplyLayout` not only applies layout but also set layout if `initial_layout` is supplied as its constructor's argument. Users can apply their own valid layout just running ApplyLayout pass. Some may think `ApplyLayout` do the things more than what the name implies. 

I'll follow the majority choice. (I personally like option B.)
Which is the choice of reviewers? (Or another option?)

Reply:
@1ucian0 @ajavadia Thoughts?

Reply:
I would prefer A, because it is more clear about the flow (Option B is more general, but it blurs and complicates the flow a bit).

The flow I have in my mind is:
```
(Analysis)               SetLayout        NoiseAdaptiveLayout         DenseLayout

                                \                  |                  /

(Transformation)                               ApplyLayout

                                /                  |                  \

(Transformation)          BasicSwap           StochasticSwap          LookaheadSwap
```

Reply:
(^ Maybe `SetLayout` should be called "ManualLayout" or "UserLayout" to be clear it is different from `ApplyLayout`.)

Reply:
OK. I'll follow Option A. I remind the philosophy, keep each pass simple, and provide useful functions by their combination. A preset passmanager with SetLayout + ApplyLayout would be useful for users who just want to run a simple circuit on physical qubits they select.
I prefer ManualLayout to SetLayout, but leave it as is in this PR.